File: core/src/main/java/ru/pocketbyte/locolaser/exception/InvalidConfigException.java
/*
 * Copyright © 2017 Denis Shurygin. All rights reserved.
 * Licensed under the Apache License, Version 2.0
 */

package ru.pocketbyte.locolaser.exception;

/**
 * @author Denis Shurygin
 */
public class InvalidConfigException extends Exception {
    private static final String DEFAULT_MESSAGE = "Invalid Config!";

    public InvalidConfigException() {
        this(null);
    }

    public InvalidConfigException(String message) {
        super(message != null ? DEFAULT_MESSAGE + " " + message : DEFAULT_MESSAGE);
    }
}

File: core/src/main/java/ru/pocketbyte/locolaser/exception/InvalidConfigException.kt
/*
 * Copyright © 2017 Denis Shurygin. All rights reserved.
 * Licensed under the Apache License, Version 2.0
 */

package ru.pocketbyte.locolaser.exception

/**
 * Exception thrown when a localization config is invalid or cannot be processed.
 *
 * @param message Optional detail message appended to the default error message.
 * @author Denis Shurygin
 */
class InvalidConfigException @JvmOverloads constructor(
        message: String? = null
) : Exception(if (message != null) "$DEFAULT_MESSAGE $message" else DEFAULT_MESSAGE) {
    companion object {
        private const val DEFAULT_MESSAGE = "Invalid Config!"
    }
}
